nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/engine/StatelessFlowManager.java

```diff
@@ -296,6 +298,53 @@ public ReportingTaskNode createReportingTask(final String type, final String id,
         return taskNode;
     }
 
+    @Override
+    public ParameterProviderNode createParameterProvider(final String type, final String id, final BundleCoordinate bundleCoordinate, final Set<URL> additionalUrls, final boolean firstTimeAdded,
```

Comment:
Within Stateless, we have a ParameterValueProvider, which is capable of fetching parameter values. This notion of a ParameterProvider is a bit different, where users have the ability to go fetch parameters and apply them. I'm not sure we need to support this at all, TBH, in stateless. Can probably just have this `throw new UnsupportedOperationException();` no?

Reply:
I agree some more careful thought is needed regarding these two concepts.  For this PR, I'll just throw an `UnsupportedOperationException` here.